samanthas_telegram_bot/conversation/auxil/message_sender.py

```diff
@@ -0,0 +1,176 @@
+# This module contains some send_message operations that are too complex to be included in the main
+# code, and at the same time need to run multiple times.
+import datetime
+from contextlib import suppress
+
+import telegram.error
+from telegram import (
+    InlineKeyboardButton,
+    InlineKeyboardMarkup,
+    KeyboardButton,
+    ReplyKeyboardMarkup,
+    Update,
+)
+from telegram.constants import ParseMode
+
+from samanthas_telegram_bot.conversation.constants_enums import (
+    PHRASES,
+    CommonCallbackData,
+    ConversationMode,
+    Role,
+)
+from samanthas_telegram_bot.conversation.custom_context_types import CUSTOM_CONTEXT_TYPES
+
+
+class MessageSender:
+    """A helper class that sends replies to user by executing
+    `update.effective_chat.send_message()`.
+
+    Methods in this class are called several times in the bot's code and/or are complex.
+
+    All methods in this class are **static methods**. This is a class is just a namespace
+    and a way to be stylistically consistent with other helper classes.
+    """
+
+    @staticmethod
+    async def ask_phone_number(update: Update, context: CUSTOM_CONTEXT_TYPES) -> None:
+        """Sends a message to ask for phone number."""
+        await update.effective_chat.send_message(
+            PHRASES["ask_phone"][context.user_data.locale],
+            disable_web_page_preview=True,  # the message contains link to site with country codes
+            parse_mode=ParseMode.MARKDOWN_V2,
+            reply_markup=ReplyKeyboardMarkup(
+                [
+                    [
+                        KeyboardButton(
+                            text=PHRASES["share_phone"][context.user_data.locale],
+                            request_contact=True,
+                        )
+                    ]
+                ]
+            ),
+        )
+
+    @staticmethod
+    async def ask_review(update: Update, context: CUSTOM_CONTEXT_TYPES) -> None:
```

Comment:
this functions looks very complex. I think it's possible to split it into smaller functions

Reply:
OK, factored out functionality that constructs the message.